src/vector_index.c

```diff
@@ -38,8 +38,9 @@ VecSimIndex *openVectorIndex(IndexSpec *spec, RedisModuleString *keyName, bool c
   // create new vector data structure
   kdv = rm_calloc(1, sizeof(*kdv));
   kdv->p = VecSimIndex_New(&fieldSpec->vectorOpts.vecSimParams);
-
-  dictAdd(spec->keysDict, keyName, kdv);
+  if(kdv->p){
+    dictAdd(spec->keysDict, keyName, kdv);  
+  }
```

Comment:
Why did we added this change? This is a leak of `kdv` first of all, and also we will try to create the index every time (and fail every time) this function is called with the create flag

Reply:
It's a solution for a bug,
In flushall, it calls valFreeCb (spec.c, 1882) that frees each key in kdv, which causes a crash if the index pointer is added to kdv even tough it's null.
I think it's ok, cause if kdv->p is null it means the index couldn't be opened so it's ok we try (and fail) again.

Reply:
OK, I agree it's fine to retry and fail to create each time in these rare cases.
But we do need to free the `kdv` allocation if we don't add it to the dictionary (or avoid allocating it)